src/main/java/Model/Database/Interaction/I_ControllerUnit.java

```diff
@@ -217,6 +218,73 @@ public static T_ControllerUnit retrieveByUid(Connection conn, PreparedStatement
         return arr;
     }
 
+    /***
+     *
+     * @param conn
+     * @param ps
+     * @param rs
+     * @return
+     * @throws SQLException
+     */
+    public static List<T_ControllerUnit> retrieveFilteredAll(Connection conn, PreparedStatement ps, ResultSet rs, int centralUnitId, int flatId, String dip) throws SQLException {
+
+        // No Filter is being used
+        if (centralUnitId <= DB_DO_NOT_USE_THIS_FILTER && flatId  <= DB_DO_NOT_USE_THIS_FILTER) {
+            return retrieveAll(conn, ps, rs);
+        }
+
+        // SQL Definition
+        String usedSql = "SELECT " +
+                "* " +
+                "FROM " + T_ControllerUnit.DBTABLE_NAME + " " +
+                "WHERE ";
+
+
+        // add filter rules
+        boolean centralRule = centralUnitId > 0;
+        boolean flatRule = flatId > 0;
+        boolean dipRule = dip.equals("") == false;
+
+        usedSql = (centralRule ? usedSql + T_ControllerUnit.DBTABLE_NAME + ".CentralUnitID=? " : usedSql);
```

Comment:
fu kamo je som prekomplikovany typ ale toto je aj namna moc. ked funguje kasli nato ale aspon pridaj komentar priklad vsetky moznosti ktore mozu nastat

Reply:
tu je moznost ze to budes chciet nefiltrovane ked zadas bud prazdny dip alebo flatRule napr. -1 a centralUnitId -1? ked by si vsetky takto nastavil, nie je to ze proste mozes zavolat nefiltrovanu metodu co ti zavola vsetko alebo je to takto aby sme nemuseli mat dve metody - jednu pre vsetko a jednu pre filtrovane, ale rozlisis to tym co zadas do filtrovania? filtre alebo ziadne filtre? toto urcite treba zdokumentovat